source/adios2/toolkit/derived/parser/ASTNode.cpp

```diff
@@ -0,0 +1,201 @@
+#ifndef ADIOS2_DERIVED_PARSER_ASTNODE_CPP_
+#define ADIOS2_DERIVED_PARSER_ASTNODE_CPP_
+
+#include "ASTNode.h"
+
+namespace adios2
+{
+namespace detail
+{
+
+/*****************************************/
+// alias maps to pair of path and indices (indices may be empty string)
+std::map<std::string, std::pair<std::string, std::string>> ASTNode::var_lookup;
+int ASTNode::num_expr = 0;
```

Comment:
Can num_expr go?  Leftover from debugging?

Reply:
Yes, `num_expr` is only used for debugging, I'll remove it from everywhere it's used. Everything in the parser is Liz's code (I'm only the commiter for that commit not the author) but for some reason git doesn't like her address. I tried not to remove too much from her codes since I am not sure what she plans to use for more complex expressions but we can always put it back.